src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs

```diff
@@ -935,6 +935,213 @@ public static void ServiceContract_TypedProxy_SyncOperation_UniqueTypeRefArg()
         }
     }
 
+    [Fact]
+    [OuterLoop]
+    public static void BasicHttp_Async_Open_ChannelFactory()
+    {
+        string testString = "Hello";
+        BasicHttpBinding binding = null;
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
+        string result = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+            binding.OpenTimeout = ScenarioTestHelpers.TestTimeout;
+            factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+            // *** EXECUTE *** \\
+            Task t = Task.Factory.FromAsync(factory.BeginOpen, factory.EndOpen, TaskCreationOptions.None);
+
+            // *** VALIDATE *** \\
+            t.GetAwaiter().GetResult();
```

Comment:
Ah, I think I see why. I think the next test wouldn't actually test that everything happens on a different thread if this test wasn't blocking.

---
In reply to: [60987684](https://github.com/dotnet/wcf/pull/1071#discussion_r60987684) [](ancestors = 60987684)